src/BoxAnnotations.ts

```diff
@@ -0,0 +1,157 @@
+    file?: {
+        permissions?: Permissions;
+    };
+    viewer?: {
+        NAME: string;
+    };
+};
+
+type ViewerConfig = {
+    enabled?: boolean;
+    enabledTypes?: string[];
+};
+
+type ViewerOption = {
+    enabledTypes?: string[];
+};
+
+type ViewerOptions = {
+    [index: string]: ViewerOption;
+    Document: ViewerOption;
+};
+
+/**
+ * NAME: The name of the annotator.
+ * CONSTRUCTOR: Constructor for the annotator.
+ * VIEWERS: The kinds of viewers that can be annotated by this.
+ * TYPES: The types of annotations that can be used by this annotator.
+ */
+const ANNOTATORS: Annotator[] = [
+    {
+        NAME: 'Document',
+        CONSTRUCTOR: DocAnnotator,
+        VIEWERS: ['Document', 'Presentation'],
+        TYPES: [TYPES.draw, TYPES.highlight, TYPES.highlight_comment, TYPES.point],
+    },
+];
+
+const ANNOTATOR_TYPE_CONTROLLERS = {
+    [TYPES.point]: {
+        CONSTRUCTOR: PointModeController,
+    },
+    [TYPES.highlight]: {
+        CONSTRUCTOR: HighlightModeController,
+    },
+    [TYPES.highlight_comment]: {
+        CONSTRUCTOR: HighlightModeController,
+    },
+    [TYPES.draw]: {
+        CONSTRUCTOR: DrawingModeController,
+    },
+};
+
+class BoxAnnotations {
+    annotators: Annotator[];
+
+    viewerOptions: ViewerOptions;
+
+    /**
+     * [constructor]
+     *
+     * @param {Object} viewerOptions - Viewer-specific annotator options
+     * @return {BoxAnnotations} BoxAnnotations instance
+     */
+    constructor(viewerOptions: ViewerOptions) {
+        this.annotators = ANNOTATORS;
+        this.viewerOptions = viewerOptions || {};
+    }
+
+    canLoad(permissions?: Permissions): boolean {
+        if (!permissions) {
+            return false;
+        }
+
+        return (
+            !!permissions[PERMISSIONS.CAN_ANNOTATE] ||
+            !!permissions[PERMISSIONS.CAN_VIEW_ANNOTATIONS_ALL] ||
+            !!permissions[PERMISSIONS.CAN_VIEW_ANNOTATIONS_SELF]
+        );
+    }
+
+    instantiateControllers(annotator: Annotator): Annotator | null {
+        if (!annotator || !annotator.TYPES || annotator.CONTROLLERS) {
+            return annotator;
+        }
+
+        annotator.TYPES.forEach(type => {
+            if (type in ANNOTATOR_TYPE_CONTROLLERS) {
+                annotator.CONTROLLERS = annotator.CONTROLLERS || {};
+                annotator.CONTROLLERS[type] = new ANNOTATOR_TYPE_CONTROLLERS[type].CONSTRUCTOR(annotator.NAME);
+            }
+        });
+
+        return annotator;
+    }
+
+    /**
+     * Returns the available annotators
+     *
+     * @return {Array} List of supported annotators
+     */
+    getAnnotators(): Annotator[] {
+        return Array.isArray(this.annotators) ? this.annotators : [];
+    }
+
+    /**
+     * Get all annotators for a given viewer.
+     *
+     * @param {string} viewerName Name of the viewer to get annotators for
+     * @return {Object} Annotator for the viewer
+     */
+    getAnnotatorsForViewer(viewerName = ''): Annotator | undefined {
```

Comment:
Do we not need to worry about disabled viewers because we're starting fresh?

Reply:
The old code supported an evidently-deprecated "disabled annotators" option via the viewer config for backward compatibility. I think it should be safe to remove it in favor of allowing the host to specify enabled viewers, instead. What do you think.

Reply:
I'm fine with removing it if it's not supported anymore. Is the mechanism then to specify in the viewer options?

Reply:
Yeah, you can specify what's enabled either in viewer options or in the config object that's passed into the entry constructor.